openid_connect_configuration.conf

```diff
@@ -28,6 +28,27 @@ map $host $oidc_jwt_keyfile {
     default "http://127.0.0.1:8080/auth/realms/master/protocol/openid-connect/certs";
 }
 
+map $host $oidc_logout_endpoint {
+    default "http://127.0.0.1:8080/auth/realms/master/protocol/openid-connect/logout";
+}
+
+map $host $oidc_logout_query_params_option {
+    # 0: default query params for the RP-initiated logout
+    # 1: extra query params is added after the default query params
+    # 2: replace default query params with custom query params
+    default 0;
+}
+
+map $host $oidc_logout_query_params {
```

Comment:
- Yes, I am aware of it.
- I have suggested the naming change in the https://github.com/nginxinc/nginx-openid-connect/issues/77 for consistency between NGINX Plus and NGINX Management Suite.
- I wanted to synchronize the variable name as it has been recently merged if possible between NGINX Plus and [NGINX Management Suite](https://docs.nginx.com/nginx-management-suite/admin-guides/access-control/configure-authentication/#enable-oidc).
  - NGINX Plus OIDC: `$oidc_authz_extra_args` is merged (Dec/8/2022)
  - NGINX Management Suite: `$oidc_authz_query_params` is released (Jul/20/2022)

Reply:
Also, I wanted to follow the naming of `query params` like the [OpenAPI spec](https://swagger.io/docs/specification/describing-parameters/) and [IETF HTTP Spec](https://www.ietf.org/archive/id/draft-ietf-httpbis-safe-method-w-body-02.html) instead of `args`.
Please let me know what you want.

Reply:
- It makes sense to use `extra_query_params` rather than `extra_args` although there is no required query parameters declared for the `$oidc_end_session_endpoint`.
- FYI. The required query parameters declared in `authZArgs` for the authz endpoint are not fully enough to be reconfigurable by customers unless they change codes. Hence, we can think of how the required fields' value can be configurable in the future. (Therefore, I used three ways of`extra`, `replace` and `built-in`. )

To sum up: changed it as `extra_query_params` at this moment.

Reply:
It looks like you've lost the thread of the discussion:

1. In the original iteration, you proposed the `$oidc_logout_query_params` var, which involved modifying only a subset of all query parameters.
2. For consistency with `$oidc_authz_extra_args`, I suggested to use to `$oidc_logout_extra_args`.
3. You referred to the fact that in your product (NGINX Management Suite) you already use the `$oidc_authz_query_params` var name and in general the standard says that these arguments are called query parameters.
4. I pointed out at that moment that we are not proposing to modify all the query parameters, but only part of them, then the variable should be called ...extra_query_params.
5. Since in the current iteration you proposed to be able to change ALL query parameters for logout uri, well, call it `...query_params`.

Reply:
Thanks for the notes. I have addressed the comments. Let me know if I miss anything.